Throw an error if point used to determine halfspace lies on surface (#4)
kkiesling/moose: framework/doc/content/source/csg/CSGBase.md

const auto & surface = csg_obj->addSurface(surf_ptr);
```

!alert! note title=Adding surfaces to the CSGBase instance

> Surfaces need to be added to the CSGBase instance in the manner listed above. If this is not done and these surfaces are referenced in regions used to define cells within the CSGBase instance, an error will occur.

Reviewer: ```suggestion
Surfaces need to be added to the CSGBase instance with `addSurface` as described above. If this is not done and these surfaces are referenced in regions used to define cells within the CSGBase instance, an error will occur.
```
